src/Components/Facility/Consultations/LogUpdateAnalayseTable.tsx

```diff
@@ -0,0 +1,127 @@
+import React from "react";
+import { formatDate, formatTime } from "../../../Utils/utils";
+import { classNames } from "../../../Utils/utils";
+import { useTranslation } from "react-i18next";
+
+interface SharedSectionTableProps {
+  data: Record<string, any>;
+  rows: Array<{ title?: string; field?: string; subField?: boolean }>;
+  choices?: Record<string, Record<number | string, string>>;
+}
+
+const LogUpdateAnalayseTable: React.FC<SharedSectionTableProps> = ({
+  data,
+  rows,
+  choices = {},
+}) => {
+  const { t } = useTranslation();
+
+  // Helper function to get the display value
+  const getDisplayValue = (
+    value: string | boolean | null | undefined,
+    field?: string,
+  ): string => {
+    if (value == null) {
+      return " ";
+    }
+
+    if (typeof value === "boolean") {
+      return t(value ? "yes" : "no");
+    }
+    if (field && choices[field]) {
+      const choice =
+        choices[field][value as keyof (typeof choices)[typeof field]];
+      return choice ? t(`${field.toUpperCase()}__${choice}`) : "-";
+    }
+    if (value && typeof value == "string") return value;
+
+    return "-";
+  };
+
+  const isValidDate = (str: string) => {
+    let ct = 0;
+    for (let i = 0; i < str.length; i++) {
+      if (str[i] == "/") ct++;
+    }
+
+    if (ct == 2) return true;
+    return false;
+  };
+
+  const dateConversion = (str: string) => {
+    const time = str.split(";")[0].trim();
+
+    const date = str.split(";")[1].trim();
+
+    const dd = date.split("/")[0];
+    const mm = date.split("/")[1];
+
+    const yyyy = date.split("/")[2];
+
+    return time + ";" + mm + "/" + dd + "/" + yyyy;
+  };
```

Comment:
we can use `formatDateTime` instead of this right?

Reply:
@rithviknishad actually formatDateTime uses days.js that expects date in format mm/dd/yyyy but the date format passed in nursingtab was different so I had to convert the format and then pass it to formatDateTime function

Reply:
From ChatGPT

In this case, you can utilize `day.js` to handle the date format transformation easily. Here's a suggestion for tackling the date formatting using `day.js`:

1. You can parse the incoming date string using `day.js`, and then transform it to the format required by `formatDateTime`.

Here’s a sample implementation for transforming the date format:

```ts
import dayjs from 'dayjs';

const dateConversion = (str: string) => {
  const [time, date] = str.split(';').map(s => s.trim());

  // Assuming the incoming date is in "dd/mm/yyyy" format
  const parsedDate = dayjs(date, 'DD/MM/YYYY');
  
  // Format the date to the required "mm/dd/yyyy" format for formatDateTime
  const formattedDate = parsedDate.format('MM/DD/YYYY');

  return `${time}; ${formattedDate}`;
};
```

In this implementation:

- The `date` is first split and trimmed.
- The `dayjs(date, 'DD/MM/YYYY')` parses the input date, assuming the format `DD/MM/YYYY`.
- The `.format('MM/DD/YYYY')` converts it into the required `MM/DD/YYYY` format for `formatDateTime`.

You can now pass the result to `formatDateTime` easily after this conversion.

Reply:
backend is giving the data in standard format @Sahil-Sinha-11 

<img width="1708" alt="image" src="https://github.com/user-attachments/assets/49309aaf-e4b1-44e2-ad4f-dcee4f9de564">

However you are already formatting it when sending the data to the component after receiving it from the backend, and then deconstructing the formatted back to unformatted, just for format it again?

<img width="726" alt="image" src="https://github.com/user-attachments/assets/c00d77b9-7451-420b-964d-a9fde6655902">

Reply:
@rithviknishad I am not able to run the application it shows a file does not exist( pluginMap). I even tried on devlop branch.
<img width="1710" alt="Screenshot 2024-10-21 at 10 30 25 PM" src="https://github.com/user-attachments/assets/d021ae31-54a7-4ab2-8729-b66fda96e217">.

Reply:
https://github.com/ohcnetwork/care_fe#install-the-required-dependencies

Reply:
Hey @rithviknishad , I resolved it .